src/Hamiltonians/correlation_functions.jl

```diff
@@ -68,22 +97,98 @@ function diagonal_element(::G2RealCorrelator{D}, add::CompositeFS) where {D}
     M = num_modes(add)
     d = mod(D, M)
     v = sum(map(onr, add.components))
-    result = 0
-    for i in eachindex(v)
-        result += v[i] * v[mod1(i + d, M)]
-    end
-    return result / M
+    return circshift_dot(v, v, (d,)) / M
 end
 
 num_offdiagonals(::G2RealCorrelator, ::SingleComponentFockAddress) = 0
 num_offdiagonals(::G2RealCorrelator, ::CompositeFS) = 0
 
-# not needed:
-# get_offdiagonal(::G2RealCorrelator, add)
-# starting_address(::G2RealCorrelator)
+"""
+    G2RealSpace(::Geometry, σ=1, τ=1; sum_components=false) <: AbstractHamiltonian{SArray}
+
+Two-body operator for density-density correlation for all displacements ``d`` in the
+specified geometry.
```

Comment:
This does not yet explain the vector character of this operator very well. What are "all displacements"? Would it be useful to link to `Offsets` here?